geode-redis/src/distributedTest/java/org/apache/geode/redis/internal/executor/hash/HlenDUnitTest.java

```diff
@@ -0,0 +1,169 @@
+  private static ClientResources resources;
+
+  @BeforeClass
+  public static void classSetup() {
+    redisPorts = AvailablePortHelper.getRandomAvailableTCPPorts(3);
+
+    String redisPort1 = String.valueOf(redisPorts[0]);
+    String redisPort2 = String.valueOf(redisPorts[1]);
+
+    locator = cluster.startLocatorVM(0);
+
+    server1 = startRedisVM(1, redisPorts[0]);
+    server2 = startRedisVM(2, redisPorts[1]);
+
+    DUnitSocketAddressResolver dnsResolver =
+        new DUnitSocketAddressResolver(new String[] {redisPort2, redisPort1});
+
+    resources = ClientResources.builder().socketAddressResolver(dnsResolver).build();
+
+    RedisClient redisClient = RedisClient.create(resources, "redis://localhost");
+    redisClient.setOptions(ClientOptions.builder().autoReconnect(true).build());
+
+    connection = redisClient.connect();
+    lettuce = connection.sync();
+  }
+
+  private static MemberVM startRedisVM(int vmID, int redisPort) {
+    int locatorPort = locator.getPort();
+
+    return cluster.startRedisVM(vmID, (x) -> x
+        .withConnectionToLocator(locatorPort)
+        .withProperty(REDIS_PORT, "" + redisPort));
+  }
+
+  @Before
+  public void testSetup() {
+    lettuce.flushall();
+  }
+
+  @AfterClass
+  public static void tearDown() throws Exception {
+    resources.shutdown().get();
+    connection.close();
+
+    server1.stop();
+    server2.stop();
+  }
+
+  @Test
+  public void testConcurrentHLens_returnExpectedLength() {
+    AtomicLong client1Len = new AtomicLong();
+    AtomicLong client2Len = new AtomicLong();
+
+    String key = "HLEN";
+
+    Map<String, String> setUpData = makeInitialHashMap(HASH_SIZE, "field", "value");
+
+    lettuce.hset(key, setUpData);
+
+    new ConcurrentLoopingThreads(NUM_ITERATIONS,
+        i -> {
+          long len = lettuce.hlen(key);
+          client1Len.addAndGet(len);
+        },
+        i -> {
+          long len = lettuce.hlen(key);
+          client2Len.addAndGet(len);
+        })
+            .run();
+
+    assertThat(client1Len.get() + client2Len.get()).isEqualTo(NUM_ITERATIONS * HASH_SIZE * 2);
+  }
+
+  @Test
+  public void testConcurrentHLen_whileAddingFields() {
+    String key = "HLEN";
+    String storeKey = "storedLength";
+
+    Map<String, String> setUpData =
+        makeInitialHashMap(HASH_SIZE, "filler-", String.valueOf(HASH_SIZE));
+    lettuce.hset(key, setUpData);
+
+    new ConcurrentLoopingThreads(NUM_ITERATIONS,
+        (i) -> {
+          int newElementCount = i + 1; // convert index to a length
+          int currentLength = HASH_SIZE + newElementCount;
+
+          lettuce.hset(key, "field-" + currentLength, String.valueOf(currentLength));
+          lettuce.set(storeKey, String.valueOf(currentLength));
+        },
+        (i) -> {
+          long actualLength = lettuce.hlen(key);
+          long expectedLength = Long.parseLong(lettuce.get(storeKey));
+
+          assertThat(actualLength).isGreaterThanOrEqualTo(expectedLength);
+        }).run();
```

Comment:
I'm wondering, since this is concurrent, if the second function might try to get the `storeKey` before the `currentLength` value has been set, resulting in that `java.lang.NumberFormatException: null` error we see in the DUnit tests.  Maybe we could do: `lettuce.set(storeKey, String.valueOf(HASH_SIZE))` before we run the `ConcurrentLoopingThreads` so there would definitely be a value there.

Reply:
great suggestion, thanks!

Reply:
that distributed test failure is exactly this bug! resolved in latest commit